source/updateCheck.py

```diff
@@ -222,10 +221,8 @@ def _executeUpdate(destPath):
 		else:
 			executeParams = u"--launcher"
 	# #4475: ensure that the new process shows its first window, by providing SW_SHOWNORMAL
-	shellapi.ShellExecute(None, None,
-		destPath,
-		executeParams,
-		None, winUser.SW_SHOWNORMAL)
+	core.NewNVDAInstanceOnExitManager.queueInstance(destPath, executeParams)
+	core.triggerNVDAExit()

```

Comment:
I don't really like the coupling between `NewNVDAInstanceOnExitManager`, `triggerNVDAExit` and `preNVDAExit`. 

Looking at this again, it seems the `preNVDAExit` action doesn't do what it claims. If an add-on wanted to be notified before NVDA exited, and tried to use this action, they would be called AFTER all the modules had been terminated.

I think this PR has helped to point out the issues, but lets have a discussion on Monday about the overall NVDA shutdown approach.

I think we have a good handle on the requirements now.